chunk/tag.go

```diff
@@ -216,3 +218,57 @@ func decodeInt64Splice(buffer *[]byte) int64 {
 	*buffer = (*buffer)[n:]
 	return val
 }
+
+// marshall friendly tag structure
+type jsonTag struct {
```

Comment:
let's just make the `Address` field a string field on `Tag` and remove custom json code. 
or define `(Un)MarshalJSON` on `Address` type.

Reply:
Done

Reply:
no. I mean on the `Tag`, the exported fields should be exactly these, so you can just JSON marshal it without using any custom marshaller.

Reply:
did you not mean making "Address" in Tag to string?

Reply:
He meant to have the fields on `Tag` exported, then add a custom json marshal/unmarshal on `chunk.Address`, that way the built in json marshaler will serialize/deserialize the exported fields, but will pick up the `Address` type and call the custom marshaler function you define on it to convert from/to string

Reply:
@zelig an idea about how to serialize the response:
```js
{
  tag: id
  stored: {
    remaining: 100
    total: 1000
  }
  split: {
    remaining: 100
    total: 1000
  }
  ...
}
```
if we just serialize the struct as is, we end up delegating the logic which is in `Tag.Status` to the client:
```go
// Status returns the value of state and the total count
func (t *Tag) Status(state State) (int64, int64, error) {
	count, seen, total := t.Get(state), atomic.LoadInt64(&t.seen), atomic.LoadInt64(&t.total)
	if total == 0 {
		return count, total, errNA
	}
	switch state {
	case StateSplit, StateStored, StateSeen:
		return count, total, nil
	case StateSent, StateSynced:
		stored := atomic.LoadInt64(&t.stored)
		if stored < total {
			return count, total - seen, errNA
		}
		return count, total - seen, nil
	}
	return count, total, errNA
}
```
i am not sure that the approach of serializing the struct as is is the way to go as of such.

Reply:
@zelig ... I did exported the fields of Tags and @acud didn't like that design. He is of the opinion that we should not export fields of Tag.  He wanted to have a separate JSON Marshaller and UnMarsheller.

Reply:
> @zelig ... I did exported the fields of Tags and @acud didn't like that design. He is of the opinion that we should not export fields of Tag. He wanted to have a separate JSON Marshaller and UnMarsheller.

I all for code simplicity,  the tag  struct contains all the info, we dont need to second guess what the renderer will massage on the data..
@acud if you insist, we  could have  it your way.

Anyhow we should continue  with http upload responses using the  tags

Reply:
ok. I am exporting all the fields of Tags and closing it for now.